src/clj/game/core/tags.clj

```diff
@@ -33,7 +33,11 @@
        (trigger-event state :runner :tags-changed {:new-total new-total
                                                    :old-total old-total
                                                    :is-tagged is-tagged?}))
-     changed?)))
+     (if is-tagged?
+       (wait-for
+         (trash-on-tag state nil)
```

Comment:
this doesn't work because the calling context isn't async. it will return nil to fake-checkpoint which will loop. better to just call the function with a new eid and not worry about the async context, if this really is the best place for this check.

Reply:
Fair. I'm not particularly sure where else we can throw it - the idea is this meta-function gets called every fake checkpoint (up to 10 times), so this is guaranteed to enforce the trash condition on those cards.
That did work, but only because we just threw away the result of trash-on-tag (and wait-for) anyway and returned the changed? fn we had before.

Seems like we're still on all tests passing though :100: